### DISCUSSION TITLE
Problem on sending udp message to node from python

### DISCUSSION THREAD

**Discussion by aceylan** [contiki-ng/contiki-ng#2782] [Q&A]
Hi all!
I made an example. There is a border router and udp motes. 
My mote's codes are:
(packet generation is not important for now, just for another project)
```
#include "contiki.h"
#include "sys/node-id.h"
#include "lib/random.h"
#include "net/ipv6/uip-ds6.h"
#include "net/ipv6/simple-udp.h"
//use rpl lite
#include "net/routing/rpl-lite/rpl.h"
#include "net/routing/routing.h"

#include <inttypes.h>
#include "sys/log.h"
#define LOG_MODULE "App"
#define LOG_LEVEL LOG_LEVEL_INFO

#define UDP_PORT	8765

static void
udp_rx_callback(struct simple_udp_connection *c,
         const uip_ipaddr_t *sender_addr,
         uint16_t sender_port,
         const uip_ipaddr_t *receiver_addr,
         uint16_t receiver_port,
         const uint8_t *data,
         uint16_t datalen) // udp ile gelen paketleri al
{
  LOG_INFO("Message received\n");
  uint32_t seqnum; 
  memcpy(&seqnum, data, sizeof(seqnum)); 
  LOG_INFO("app receive packet seqnum=%" PRIu32 " from=", seqnum);
  LOG_INFO_6ADDR(sender_addr); 
  LOG_INFO_("\n"); 
}

PROCESS(node_process, "Node"); // node_process adında bir process oluştur
AUTOSTART_PROCESSES(&node_process); // node_process i otomatik başlat
/*---------------------------------------------------------------------------*/
PROCESS_THREAD(node_process, ev, data) // node_process thread i
{
  static struct etimer periodic_timer; 
  static uint32_t seqnum; 
  static struct simple_udp_connection udp_conn;
  uip_ipaddr_t dest_address; 

  PROCESS_BEGIN();

  simple_udp_register(&udp_conn, UDP_PORT, NULL,
                      UDP_PORT, udp_rx_callback);

  
    /* initialize the packet generation timer */
    etimer_set(&periodic_timer, APP_WARM_UP_PERIOD_SEC * CLOCK_SECOND
        + random_rand() % (APP_SEND_INTERVAL_SEC * CLOCK_SECOND));
//APP_WARM_UP_PERIOD_SEC * CLOCK_SECOND + random_rand() % (APP_SEND_INTERVAL_SEC * CLOCK_SECOND)
    /* wait for the warm-up period to expire */
    while(1) {
      PROCESS_WAIT_EVENT_UNTIL(etimer_expired(&periodic_timer));

      LOG_INFO("process calisti \n");
     
      if(NETSTACK_ROUTING.node_is_reachable() && NETSTACK_ROUTING.get_root_ipaddr(&dest_address)) {
        LOG_INFO("node is reachable \n");
        //destination address is ff02::1a
        uip_ip6addr(&dest_address, 0xff02, 0, 0, 0, 0, 0, 0, 0x1a);

        LOG_INFO("node_id=%u\n", node_id);
        LOG_INFO("root_ipaddr=");
        LOG_INFO_6ADDR(&dest_address);
        LOG_INFO_("\n");
        seqnum++;
        LOG_INFO("app generate packet seqnum=%" PRIu32 " node_id=%u\n", seqnum, node_id);
        simple_udp_sendto(&udp_conn, &seqnum, sizeof(seqnum), &dest_address);
      }

      etimer_set(&periodic_timer, APP_SEND_INTERVAL_SEC * CLOCK_SECOND);
    }
  
  PROCESS_END();
}
/*---------------------------------------------------------------------------*/
```

and my python code is:
```
import socket

UDP_IP = "fd00::205:5:5:5"
UDP_PORT = 8765
MESSAGE = "Hello, world!"
sock = socket.socket(socket.AF_INET6, socket.SOCK_DGRAM)

for i in range(0, 10):
    sock.sendto(MESSAGE.encode(), (UDP_IP, UDP_PORT))

```

fd00::205:5:5:5 is getting the message. But in console, that says in csma "not for us"
I dont know why. Anyone has idea?

**Comment by joakimeriksson** [comment]
Exactly how is that node's address set up? 

CSMA will drop if the target MAC address is not its own address?

**Reply by ceylanim** [reply]
Thank you for reply. 
If possible, could you go into more depth about it? Perhaps due to my limited understanding of networking concepts, I'm wondering if the issue might be related to the MAC address, especially since both the Contiki-NG application and the Python script are running on the same machine in my setup. Is it conceivable that this might be a factor in the simulation environment (Cooja), or would such an issue also be possible in a real-world scenario with physical devices?

**Reply by joakimeriksson** [reply]
Yes, it is probably related to the MAC address. Is it a Cooja generated MAC address and a default IPv6 address you are using or do you add this address in the code somewhere?

**Reply by aceylan** [reply]
@joakimeriksson I am not using any mac address on code. I think no need to mac address with udp send(I am not sure). And IPv6 address seems on network panel on cooja. it is like fd00::202:2:2:2 . 
The python code for sending is:

import socket

```
UDP_IP = "fd00::205:5:5:5"
UDP_PORT = 8765
MESSAGE = "Hello, world!"
sock = socket.socket(socket.AF_INET6, socket.SOCK_DGRAM)

for i in range(0, 10):
    sock.sendto(MESSAGE.encode(), (UDP_IP, UDP_PORT))
```

**Reply by aceylan** [reply]
serial port output is like this:

[ERR : IPv6   ] udp: no matching connection found
[WARN: ICMPv6  ] Sending ICMPv6 ERROR message type 1 code 4 to fd00::1 from fd00::205:5:5:5
[INFO: RPL    ] creating hop-by-hop option
[INFO: CSMA   ] sending to 0001.0001.0001.0001, len 99, seqno 249, queue length 2, free packets 62
[INFO: CSMA   ] preparing packet for 0001.0001.0001.0001, seqno 248, tx 0, queue 2
[INFO: CSMA   ] tx to 0001.0001.0001.0001, seqno 248, status 0, tx 0, coll 0
[INFO: CSMA   ] packet sent to 0001.0001.0001.0001, seqno 248, status 0, tx 1, coll 0
[INFO: RPL    ] packet sent to 0001.0001.0001.0001, status 0, tx 1, new link metric 128
[INFO: CSMA   ] preparing packet for 0001.0001.0001.0001, seqno 249, tx 0, queue 1
[INFO: CSMA   ] tx to 0001.0001.0001.0001, seqno 249, status 0, tx 0, coll 0
[INFO: CSMA   ] packet sent to 0001.0001.0001.0001, seqno 249, status 0, tx 1, coll 0
[INFO: RPL    ] packet sent to 0001.0001.0001.0001, status 0, tx 1, new link metric 128
[INFO: CSMA   ] received packet from 0001.0001.0001.0001, seqno 230, len 53
[ERR : IPv6   ] udp: no matching connection found
[WARN: ICMPv6  ] Sending ICMPv6 ERROR message type 1 code 4 to fd00::1 from fd00::205:5:5:5
[INFO: RPL    ] creating hop-by-hop option
[INFO: CSMA   ] sending to 0001.0001.0001.0001, len 99, seqno 250, queue length 1, free packets 63
[INFO: CSMA   ] preparing packet for 0001.0001.0001.0001, seqno 250, tx 0, queue 1
[INFO: CSMA   ] tx to 0001.0001.0001.0001, seqno 250, status 1, tx 0, coll 0
[INFO: CSMA   ] received packet from 0001.0001.0001.0001, seqno 231, len 53
[ERR : IPv6   ] udp: no matching connection found
[WARN: ICMPv6  ] Sending ICMPv6 ERROR message type 1 code 4 to fd00::1 from fd00::205:5:5:5
[INFO: RPL    ] creating hop-by-hop option
[INFO: CSMA   ] sending to 0001.0001.0001.0001, len 99, seqno 251, queue length 2, free packets 62
[INFO: CSMA   ] received packet from 0001.0001.0001.0001, seqno 232, len 53
[ERR : IPv6   ] udp: no matching connection found
[WARN: ICMPv6  ] Sending ICMPv6 ERROR message type 1 code 4 to fd00::1 from fd00::205:5:5:5
[INFO: RPL    ] creating hop-by-hop option
[INFO: CSMA   ] sending to 0001.0001.0001.0001, len 99, seqno 252, queue length 3, free packets 61
[INFO: CSMA   ] received packet from 0001.0001.0001.0001, seqno 233, len 53
[ERR : IPv6   ] udp: no matching connection found

other motes' serial ports are:
[WARN: CSMA      ] not for us
[WARN: CSMA      ] not for us
[WARN: CSMA      ] not for us
[WARN: CSMA      ] not for us
[WARN: CSMA      ] not for us
[WARN: CSMA      ] not for us
[WARN: CSMA      ] not for us
[WARN: CSMA      ] not for us
[WARN: CSMA      ] not for us
[WARN: CSMA      ] not for us
[WARN: CSMA      ] not for us

**Comment by joakimeriksson** [comment]
I think this indicates that you might be getting an UDP packet in: [ERR : IPv6 ] udp: no matching connection found - but that it is not actually listening to that port at the moment.

**Reply by aceylan** [reply]
I think it is about sender udp ipv6 address. Maybe it doesnt work on cooja. I dont know. I will try it with real device later. Thanks for answer.
